config/volume.go

```diff
@@ -1,3 +1,31 @@
 package config
 
+import (
+	"strings"
+
+	"github.com/unweave/unweave/api/types"
+)
+
 const DefaultVolumeSize = 4
+
+// GetVolumeAttachParams reads the existing config and
+func GetVolumeAttachParams() []types.VolumeAttachParams {
+	if len(Volumes) == 0 {
+		return nil
+	}
+
+	var params = make([]types.VolumeAttachParams, 0, len(Volumes))
+	for _, volume := range Volumes {
```

Comment:
We had this before. Why not preallocate the size and use the index rather than appending?

Reply:
making a habit out of using capacitance and append is less bug prone, you don't save any extra memory allocations because the backing array is already allocated with len(Volumes) size, so it's perfectly safe to do so

Reply:
Yeah I think it's clearer to return an error because that's incorrect syntax. So the user should get an error. Since it's done in the config package, it's a little tricky with the `ui` separation we talked about. Maybe if this function call fails, the cmd package should just print "Invalid volume syntax"

Reply:
okay :+1: